cypress/integration/admin/pageBuilder/blocks/pageBlocks.cy.ts

```diff
@@ -0,0 +1,99 @@
+import { customAlphabet } from "nanoid";
+
+context("Page Builder - Blocks", () => {
+    let tokenStorage;
+    const nanoid = customAlphabet("abcdefghijklmnopqrstuvwxyz");
+    const blockCategoryName = nanoid(10); // Generate a random 10-character lowercase string
+    const slugName = nanoid(10); // Generate another random 10-character lowercase string
+
+    beforeEach(() => cy.login());
+    beforeEach(() => cy.pbDeleteBlocks());
+    beforeEach(() => cy.pbAllDeleteBlockCategories()); 
+
+    it("Should be able to create a block category and a block and then edit, duplicate and delete it", () => {
+
+        //Navigates to page, checks for proper loading and then clicks create new category button
+        cy.visit("/page-builder/page-blocks");
+        cy.findByPlaceholderText("Search blocks").should("exist");
+        cy.findByTestId("pb-blocks-list-new-block-btn").click();
+        cy.findByTestId("pb-blocks-list-new-block-category-btn").click();
+        
+        //Fills in block category data and saves it
+        cy.findByRole("textbox", { name: "Name" }).type(blockCategoryName);
+        cy.findByRole("textbox", { name: "Slug" }).type(slugName);
+        cy.findByRole("textbox", { name: "Description" }).type(blockCategoryName);
+        cy.findByTestId("pb-block-categories-form-save-block-category-btn").click();
+
+        //Check is category search bar works as expected
```

Comment:
We'll be testing search functionality in the Webiny Batch 1 document. So, I think we don't want need to test search here.